tls/s2n_post_handshake.c

```diff
@@ -167,12 +167,42 @@ S2N_RESULT s2n_post_handshake_recv(struct s2n_connection *conn)
     return S2N_RESULT_OK;
 }
 
+S2N_RESULT s2n_post_handshake_write_records(struct s2n_connection *conn, s2n_blocked_status *blocked)
+{
+    struct s2n_stuffer *message = &conn->handshake.io;
```

Comment:
For my own edification: Do post handshake messages go through the s2n_handshake->io stuffer? Definitely seems a bit odd. But I suppose that stuffer isn't needed anymore so we can re-use it?

Reply:
NewSessionTickets and any future large post-handshake messages go through the handshake.io stuffer. it's like a staging area for handshake messages. KeyUpdates don't need to because they're small and not fragmented.

Reusing it saves on connection memory over adding a new post_handshake.out stuffer, and it means that if we send post-handshake messages right after the handshake we probably won't need to allocate any new memory. That's a common use case, since NewSessionTickets are often sent immediately after the handshake.

Reply:
But this doesn't include application data right? AppData is definitely post handshake; are they not considered messages?

Reply:
Ah, I think the confusion is in the naming. "post-handshake messages" could more accurately be called "post-handshake handshake messages". They're handshake messages that occur after the handshake is complete. Application data records are not handshake messages.

See https://www.rfc-editor.org/rfc/rfc8446#section-4.6.